src/vma/dev/ring.h

```diff
@@ -330,12 +170,15 @@ class ring : public mem_buf_desc_owner
 	}
 #endif // DEFINED_SOCKETXTREME	
 
+	inline int get_if_index() { return m_if_index; }
+
 protected:
-	uint32_t		m_n_num_resources;
+	inline void set_parent(ring* parent) { m_parent = ( parent ? parent : this); }
+	inline void set_if_index(int if_index) { m_if_index = if_index; }
```

Comment:
both should be part of the constructor

Reply:
could be considered as enhancement task for future